src/components/TimeLine/TimeLine.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+
+import {TimeItem} from "./TimeItem";
+
+import {TimeLineProps, TimeLinePropTypes} from "./TimeLineProps";
+import {TimeLineContext, TimeLineContextTypes} from "./TimeLineContex";
+import {TimeLineState} from "./TimeLineState";
+
+import {ProjectInterface} from "../../data/Projects/ProjectInterface";
+import {projectsList} from "../../data/Projects/projectsDescriptionList";
+import {concat} from "../../helpers/concat";
+
+export class TimeLine extends React.Component<TimeLineProps, TimeLineState> {
+    public static propTypes = TimeLinePropTypes;
+    public static childContextTypes = TimeLineContextTypes;
+    public static readonly animationDuration = 300;
+
+    public readonly sliderDefaultClassName = "chronology-slider";
+    public readonly sliderMoveClassName = "is-move";
+
+    public state = {
+        activeProject: projectsList[projectsList.length - 1],
+        pointPosition: 0,
+        sliderClassName: this.sliderDefaultClassName,
+    };
+
+    protected timeLineItems: number [];
+
+    constructor(props) {
+        super(props);
+
+        this.timeLineItems = Array.from(Array(this.props.range.max - this.props.range.min + 1))
+            .fill(undefined)
+            .map((x, i) => this.props.range.min + i);
+    }
+
+    public getChildContext(): TimeLineContext {
+        return {
+            setNextProject: this.handlerNextProject
+        };
+    }
+
+    public componentDidUpdate() {
+        if (this.state.sliderClassName !== this.sliderDefaultClassName) {
```

Comment:
Use
```php
if(condition) {
    return;
}
// Your code here
```
instead

Reply:
You should create timer for restoring state class name where you set it to another value, not here